lib/core/src/display/shape/glyph/system.rs

```diff
@@ -0,0 +1,231 @@
+//! A main glyph system implementation.
+
+use crate::prelude::*;
+
+use crate::display::shape::glyph::font::FontId;
+use crate::display::shape::glyph::font::FontRenderInfo;
+use crate::display::shape::glyph::font::FontRegistry;
+use crate::display::shape::glyph::pen::PenIterator;
+use crate::display::shape::glyph::msdf::MsdfTexture;
+use crate::display::symbol::material::Material;
+use crate::display::symbol::shader::builder::CodeTemplate;
+use crate::display::world::*;
+use crate::system::gpu::data::texture::*;
+use crate::system::gpu::types::*;
+
+use nalgebra::Vector2;
+use nalgebra::Vector4;
+
+
+
+// =============
+// === Glyph ===
+// =============
+
+/// A glyph rendered on screen. The displayed character will be stretched to fit the entire bbox of
+/// underlying sprite.
+#[derive(Shrinkwrap)]
+pub struct Glyph {
+    context         : Context,
+    #[shrinkwrap(main_field)]
+    sprite          : Sprite,
```

Comment:
move this field up - it looks so nicer if the `#[shrinkwrap(main_field)]` is not between fields